src/pages/myproducts/myproducts.js

```diff
@@ -85,6 +88,30 @@ export const MyList = () => {
       });
   }, [debouncedSearch, selectedSubcategory]);
 
+    const handleDelete = (product) => {
+        // Generate the body necessary for BE
+        const body = {
+            name: product.name,
+            expiryDate: product.expiryDate,
+        };
+
+        // Delete the product
+        fetch(apiUrl + "/products", {
+            method: "DELETE",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify(body),
+        })
+            .then((res) => res.json())
+            .then((res) => {
+                // When product deleted, get the updated list
+               if (res.success) {
+                   setProducts(products.filter((p) => p.name !== product.name))
```

Comment:
Two products can have the same name. You should check which one to filter out based on both product name and expiry date

Reply:
Or as I said, you can re-fetch `/api/products` - then you won't have to filter anything out

Reply:
I will re-fetch then as I already done like this.